ocaml/libs/xapi-compression/xapi_gzip.ml

```diff
@@ -0,0 +1,67 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(* This is a minimal test CLI to exercise gzip for compression and
+   uncompression using the Gzip module; which in turn spawns a process via
+   Forkexecd *)
+
+module C = Cmdliner
+
+module Gzip = Xapi_compression.Make (struct let executable = "/bin/gzip" end)
```

Comment:
There is already a gzip module that does this

Reply:
I know, but I wanted to avoid the dependency given that it depends on this opam package - that would be circular.